### PR TITLE
fix: display donation forms via ajax in Give Form Block #4137

### LINKED ISSUE #4137: fix: display donation forms via ajax in Give Form Block
# Bug Report

## User Story
As a user, I want to select any donation form in the Give Form block for Gutenberg editor so that I can use Gutenberg editor.

## Current Behavior
Currently, I am unable to select older donation forms as the recent 10 donation forms are displayed in Give Form block.

## Expected Behavior
I expect to load all the donation forms based on AJAX input.

## Bug Type
- [ ] This bug describes functionality that once worked as expected in version X.X.X.
- [ ] This bug describes functionality that never worked as expected.
- [x] I am not sure whether this functionality ever worked as expected.

## Visuals
image (5)

## Acceptance Criteria
- [x] Ensure that all the donation forms are loaded via AJAX search (like chosen field) 
- [x] Testing

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR resolves #4137 

## How Has This Been Tested?
I've tested this PR as per the acceptance criteria. For more information, please check the video https://www.loom.com/share/fc190ee91060452b8646ec094ccc2b8b

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: blocks/donation-form/class-give-donation-form-block.php [hunk 1]
```diff
@@ -149,6 +150,37 @@ public function render_donation_form( $attributes ) {
...
+
+	/**
+	 * This function is used to fetch donation forms based on the chosen search in the donation form block.
+	 *
+	 * @since  2.5.3
+	 * @access public
+	 *
+	 * @return void
+	 */
+	public function block_donation_form_search_results() {
```
Comment (ravinderk) [inline blocks/donation-form/class-give-donation-form-block.php]: @mehul0810 did you check existing WP endpoint to get custom post type search results?

**Comment by mikejhale** [comment]
@mehul0810 This is including forms in Draft status to the list. This needs to only allow forms that are in Publish status when it returns the list of forms.

File: blocks/donation-form/class-give-donation-form-block.php (hunk 1 above)
Comment (mehul0810) [reply blocks/donation-form/class-give-donation-form-block.php]: @ravinderk This is not possible as of now with `WithSelect` function. So, I'll circle back on this point in future. So, I've implemented the custom AJAX logic here.

**Comment by mehul0810** [comment]
@mikejhale I've updated the code. Please review this PR and let me know your thoughts.

**Pull request merged**